dartagnan/src/main/java/com/dat3m/dartagnan/program/Program.java

```diff
@@ -92,35 +89,34 @@ public void add(Thread t) {
 		threads.add(t);
 	}
 
-    public EventCache getCache(){
-        if(cache == null){
-            cache = new EventCache(getEvents());
-        }
-        return cache;
-    }
-
-    public void clearCache(boolean clearThreadCaches){
-        if (clearThreadCaches) {
-            for (Thread t : threads) {
-                t.clearCache();
-            }
-        }
-    	cache = null;
-    }
-
     public List<Thread> getThreads() {
         return threads;
     }
 
-	public List<Event> getEvents(){
-        // TODO: Why don't we use the cache if available?
+    /**
+     * Iterates all events in this program.
+     *
+     * @return {@code cId}-ordered complete sequence of all events in this program.
```

Comment:
`globalId`, though I think the ids should not even be mentioned.
This is because the events are not ordered along the id, but the ids are assigned to match the order of the events.
If you want to be accurate, the event sequence is ordered by thread id and (syntactic) successor relationship.